src/context/AppProvider/AppContext.tsx

```diff
@@ -104,8 +105,15 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
 
   // Initial account and portfolio fetch for non-ledger wallets
   useEffect(() => {
+    const hasManagedAccounts = (() => {
+      // MM without snap doesn't allow account management - if the user just installed the snap, we know they don't have managed accounts
+      if (!previousIsSnapInstalled && isSnapInstalled) return false
+      // We know snap wasn't just installed in this render - so if there are any requestedAccountIds, we assume the user has managed accounts
+      return requestedAccountIds.length > 0
+    })()
```

Comment:
Could be even cleaner and just be a very long boolean like `Boolean(requestedAccountIds.length > 0 && (!isSnapInstalled || (!previousIsSnapInstalled && isSnapInstalled))` but IIFE to keep things cleaner and enable the commentary track so viewers get context 🎥